transcom/mymove
# B 20899 sit end date adjustments int

## [B-20899](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A993859)

## Summary

Adds the following service logic for SIT

- If SIT departure date is later than the SIT authorized end date
  - Existing tests updated to support this request (Previously SIT departure was requested without ever even having SIT entry. This broke a lot of things)
- If the SIT departure date is later than the SIT entry date but before the authorized end date
  - Then the SIT authorized end date will be updated to be equal to the departure date
  - This update will run on all service item updates
- If the SIT departure date is earlier than the SIT entry date
  - Then the request will be blocked via a validation error

### How to test

Given the Prime user is submitting updates to SIT
When the SIT departure date is earlier than the current SIT entry date,
Then no change is made, and the submitter is alerted that the SIT departure date cannot be earlier than the SIT entry date.

1. Log into prime sim
2. Select any move
3. Create new HHG shipment
4. Login as TOO
5. Approve new shipment
6. Switch back to prime sim
7. Add new origin sit service item BUT LEAVE SIT departure Date EMPTY
8. Switch back to TOO
9. Approve said service items associated with origin sit (DOFSIT sets the authorized end date)
10. Switch back to prime sim
11. Edit DOASIT
12. Set departure date to earlier than the SIT start date

Given the Prime user is submitting updates to SIT
When the SIT departure date is later than the current SIT authorized end date,
Then the SIT authorized end date will not be updated.

1. Log into prime sim
2. Select any move
3. Create new HHG shipment
4. Login as TOO
5. Approve new shipment
6. Switch back to prime sim
7. Add new origin sit service item BUT LEAVE SIT departure Date EMPTY
8. Switch back to TOO
9. Approve said service items associated with origin sit (DOFSIT sets the authorized end date)
10. Switch back to prime sim
11. Edit DOASIT
12. Set departure date to later than the current authorized end date
13. Review the database and see that the authorized end date IS NOT EQUAL TO the SIT departure date

Given a Prime user is submitting updates to SIT,
When the SIT departure date is earlier than the SIT authorized end date,
Then the SIT authorized end date will recalculate to equal the SIT departure date.

1. Log into prime sim
2. Select any move
3. Create new HHG shipment
4. Login as TOO
5. Approve new shipment
6. Switch back to prime sim
7. Add new origin sit service item BUT LEAVE SIT departure Date EMPTY
8. Switch back to TOO
9. Approve said service items associated with origin sit (DOFSIT sets the authorized end date)
10. Make note of your SIT authorized end date

![image](https://github.com/user-attachments/assets/42f70f7b-454d-4f09-96d9-9b070f8f7ab0)

11. Switch back to Prime sim
12. Edit DOASIT (Hidden behind 508 compliance violation)
13. Provide a departure date earlier than the authorized end date but after the SIT start date

![image](https://github.com/user-attachments/assets/dbd619f1-05f7-41cf-a3ca-9c596e6e0cb1)

14. Pull your shipment via db or API and make note that the origin_sit_auth_end_date will be equal to the departure date
15. Note that the UI will NOT display your fields properly at this time

![image](https://github.com/user-attachments/assets/994c1e46-c65c-424e-8c05-0e18eabc4f0f)

## Screenshots

![image](https://github.com/user-attachments/assets/95d72d6f-4fe4-48cd-8b8a-c620afe01756)
